File: Cajero automatico Java/src/com/indra/Cajero.java
package com.indra;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Cajero {

    double saldo = 1000;
    Scanner input = new Scanner(System.in);
    HashMap<String, Date> movimientos = new HashMap<>();

    public void consultar(){
        if(saldo > 0){
            System.out.println("Su saldo actual es de: " + saldo);
        }else {
            System.out.println("No cuenta con fondos");
        }
    }

    public void retirar(){
        if(saldo > 0){
            System.out.println("Ingrese la cantidad de dinero a retirar: ");
            double retiro = input.nextDouble();
            if(retiro > 0){
                if(retiro <= saldo){
                    saldo = saldo - retiro;
                    movimientos.put("Saldo anterior: " + (saldo + retiro) + " | Retiro: " + retiro + " | Saldo actual: " + saldo + " | Fecha: ", new  Date());
                    System.out.println("Su saldo actual es de: " + saldo);
                }else {
                    System.out.println("Saldo insuficiente");
                }
            }else {
                System.out.println("Ingrese una cantidad correcta");
            }


        }else {
            System.out.println("No tiene saldo a retirar");
        }
    }

    public void historico(){
        movimientos.entrySet()
                .stream()
                .sorted(Map.Entry.comparingByValue())
                .forEach(System.out::println);
    }
}

File: Cajero automatico Python/Cajero.py
import datetime

class Cajero:

    def __init__(self):
        self.saldo = 1000
        self.movimientos = {}

    def consultar(self):
        if self.saldo > 0:
            print("Su saldo actual es de: " + str(self.saldo))
        else:
            print("No cuenta con fondos")

    def retirar(self):
        if self.saldo > 0:
            print("Ingrese la cantidad de dinero a retirar: ")
            retiro = float(input())
            if retiro > 0:
                if retiro <= self.saldo:
                    self.saldo = self.saldo - retiro
                    self.movimientos["Saldo anterior: " + str(self.saldo + retiro) + " | Retiro: " + str(retiro) + " | Saldo actual: " + str(self.saldo) + " | Fecha: "] = str(datetime.datetime.now())
                    print("Su saldo actual es de: " + str(self.saldo))
                else:
                    print("Saldo insuficiente")
            else:
                print("Ingrese una cantidad correcta")


        else:
            print("No tiene saldo a retirar")

    def historico(self):
        for clave in self.movimientos:
            print(clave, self.movimientos[clave])
